racket/src/io/network/tcp-port.rkt

```diff
@@ -92,3 +95,52 @@
     [(tcp-output-port? cp)
      (set-tcp-output-port-abandon?! cp #t)
      (close-port p)]))
+
+(define max-file-offset-for-platform
+  (case (system-type 'word)
+    [(32) (sub1 (expt 2 31))]
+    [(64) (sub1 (expt 2 63))]))
+
+(define/who (tcp-copy-file dst-p path
+                           [start 0]
+                           [end (file-size path)])
```

Comment:
Using file-size seems to have race conditions, would it be better to use #f to indicate sending full file?

Reply:
Yes, that does seem like a better approach. I'll do that.